databases/week2/lesson2.sql

```diff
@@ -0,0 +1,228 @@
+
+--Delete a task
+DELETE FROM task
+WHERE id = 37;
+
+------------------------------------------
+-----------------------Part 2: School database
+--Create a new database containing the following tables:
+--Class: with the columns: id, name, begins (date), ends (date)
+--Student: with the columns: id, name, email, phone, class_id (foreign key)
+CREATE database School;
+use School;
+create table `Class`(
+`id` INT (10) UNSIGNED PRIMARY KEY AUTO_INCREMENT,
+`name` VARCHAR (255) NOT NULL,
+`begins` DATETIME,
+`ends` DATETIME 
+);
+CREATE Table `student`(
+`id` INT (10) UNSIGNED PRIMARY KEY AUTO_INCREMENT,
+`name` VARCHAR (255) NOT NULL,
+`email` VARCHAR (255) NOT NULL,
+`phone` VARCHAR(255) NULL,
+`class_id` INT (10) UNSIGNED NOT NULL,
+FOREIGN KEY (`Class_id`) REFERENCES Class(`id`) ON delete SET NULL ON update CASCADE
+);
+--Create an index on the name column of the student table.
+--add a new column to the class table named status which can only have 
+--the following values: not-started, ongoing, finished (hint: enumerations).
+CREATE index idx_name
+on student (name);
+SHOW INDEX FROM student;
+ALTER Table Class
+ADD status enum('not-started', 'ongoing', 'finished') not null;
+----------------------------------------------------------------------
+-------------------------Part 3: More queries
+use hyf_lesson2;
+
+
+--Get all the tasks assigned to users whose email ends in @spotify.com
+SELECT
+user.email,
+task.title,
+task.description
+FROM (task JOIN user_task ON task.id = user_task.task_id)
+JOIN user ON user.id = user_task.user_id
+where user.email like '%@spotify.com';
+
+--Get all the tasks for 'Donald Duck' with status 'Not started'
+SELECT
+user.name as username,
+task.title,
+status.name 
+FROM (task join user_task ON task.id = user_task.task_id) 
+JOIN user ON user.id = user_task.user_id
+join status ON status.id= task.status_id 
+where status.name = "Not started" AND user.name = 'Donald Duck';
+
+--Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+SELECT
+user.name,
+task.title,
+task.created
+FROM (task JOIN user_task ON task.id = user_task.task_id)
+JOIN user ON user.id = user_task.user_id
+WHERE month(task.created) = 9 AND user.name = "Maryrose Meadows";
+
+--Find how many tasks where created in each month, e.g. 
+--how many tasks were created in october, how many tasks were created in 
+--november, etc. (hint: use group by)
+SELECT
+month(task.created) as months,
+COUNT(task.id) as task_count
+FROM task
+GROUP BY month(task.created);
+---------------------------------------------------------------------------
+
+--------------------------------------Part 4: Creating a database
+-- create a simple university register database
+use registration;
+SET NAMES utf8mb4;
+-- database has 4 tables: student, department, course and enrollment.
+CREATE Table `student`(
+    student_id INT(10) UNSIGNED PRIMARY KEY,
+    Name VARCHAR (255) NOT NULL,
+    GPA DECIMAL(3,2)
```

Comment:
Consider renaming `Name` and `GPA` to match the casing of the other properties `name` and `gpa`